### PR TITLE
Search Results UI

### PR DESCRIPTION
**Pull request by bill-filler**
Search filter and search results panel.

### CODE REVIEW

**Comment by tuchfarber** [comment]
@bill-filler Add a `withRouter` to the HeaderContainer (see other containers for prior art), then in your doSearch method just execute ```this.props.history.push(`/search/${query}`);```

**Comment by bill-filler** [comment]
@tuchfarber 
ok. How do I get the args that I pass to /search from within components/SearchResults? Do I have to use props.match to get them?
I'll probably be passing /search/<query>/<operator>/<type>/<journalId>

**Comment by tuchfarber** [comment]
Couple of things:
1) Your path (`/x/y/z/`) should reference a page/view, parameters of that page/view should be kept in the query text, not in the path. So for this component we'll want it to just be `/search?query=w&operator=x&type=y&journalId=z`. One could argue that `journalId` should be part of the path, but I think it's easiest for it to just be a param. 
2) Once you're on the search page (`SearchResults` component), you can access the query text via the query-string library we have installed. You can access like:
```
import queryString from 'query-string';
const parsed = queryString.parse(props.location.search)
// `parsed` is now an object with the query text attributes so you can access like
const query = parsed.query
// or
const {query, operator, type, journalId} = parsed
// note variable name have to be the same as attribute names if using unpacking
```
3) When building the URL you can also use the query-string library like so:
```
const options = {
  query: 'the',
  operator: 'and',
  ...
};
const path = /search/?${queryString.stringify(options)}
```

**Comment by tuchfarber** [comment]
We're using code like `3.` in the `fetchAllJournals` method in `JournalsApiService` for example.

File: src/components/SearchResults/SearchResults.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
+$highlight-background: #f2f8fb;
+
+.search-result-card {
```
Comment (tuchfarber) [inline src/components/SearchResults/SearchResults.scss]: Can probably combine this with .card-link once you make the <Link> tag the outermost element.

File: src/components/SearchResults/SearchResults.scss [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    line-height: 1.67;
+    color: #414141;
+}
+
+.result-title {
+    padding-left: 8px;
+    font-family: OpenSans;
+    font-size: 20px;
+    font-weight: 500;
+    color: #005686;
```
Comment (tuchfarber) [inline src/components/SearchResults/SearchResults.scss]: We've already defined this color as both the `$primary` and `$deep-sea-blue`. If this is a color that should change with theming, use `$primary`, if it should always be blue (probably in this case) use `$deep-sea-blue`.

File: src/components/SearchResults/SearchResults.scss [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    text-decoration: underline;
+}
+
+.result-highlight {
+    font-family: OpenSans;
+    font-size: 16px;
+    color: #292b2c;
```
Comment (tuchfarber) [inline src/components/SearchResults/SearchResults.scss]: Can use `$dark-grey`

File: src/components/SearchResults/SearchResults.scss [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.result-breadcrumb {
+    display: inline-block;
+    font-family: OpenSans;
+    font-size: 14px;
+    font-weight: 600;
+    line-height: 1.71;
+    letter-spacing: 1px;
+    color: #636c72;
```
Comment (tuchfarber) [inline src/components/SearchResults/SearchResults.scss]: Can use `$slate-grey`

File: src/containers/SearchFilterContainer/index.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    operator: state.searchResults.operator,
+    error: state.searchResults.error,
+    searchFinished: state.searchResults.finishedFetching,
+    searchStarted: state.searchResults.startedFetching,
+  }
+);
+
+const mapDispatchToProps = dispatch => (
+  {
+    setFilter: filter =>
```
Comment (tuchfarber) [inline src/containers/SearchFilterContainer/index.jsx]: Unnecessary line break

File: src/data/actions/search.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import 'whatwg-fetch';
```
Comment (tuchfarber) [inline src/data/actions/search.js]: Not using fetch, we're using axios. We need to remove this import from other files at some point, but for now just remove this one.

File: src/data/reducers/journal.js [hunk 7]
```diff
@@ -11,7 +11,8 @@ const journal = (state = {
   longDescription: '',
   cardImageUrl: '',
   heroImageUrl: '',
-  id: null,
+  aboutId: null,
```
Comment (tuchfarber) [inline src/data/reducers/journal.js]: I think this is supposed to be `journalAboutId` to match below.

File: src/components/SearchResults/index.jsx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+      <BreadCrumb breadcrumbs={data.breadcrumbs} />
+
+    </Link>
+  </div>
+);
+
+class SearchResults extends React.Component {
+  constructor(props) {
+    super(props);
+    this.formatKey = this.formatKey.bind(this);
```
Comment (tuchfarber) [inline src/components/SearchResults/index.jsx]: NIT: You don't need to bind this to methods that aren't being passed across components.

File: src/components/SearchResults/index.jsx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    this.matchesFilter = this.matchesFilter.bind(this);
```
Comment (tuchfarber) [inline src/components/SearchResults/index.jsx]: NIT: You don't need to bind this to methods that aren't being passed across components.

**Comment by bill-filler** [comment]
@tuchfarber 
All comments from the review addressed and fixed. Thanks for the feedback.

**Pull request merged**